frontend/src/containers/AddTable.tsx

```diff
@@ -171,8 +171,27 @@ function AddTable() {
           setCurrentJson([]);
         } else {
           setCsvErrors(undefined);
-          setCsvJson(results.data);
-          setCurrentJson(results.data);
+
+          const csvJson = new Array<any>();
+          if (results.data.length) {
+            const headers = results.data[0] as Array<any>;
+            for (let i = 1; i < results.data.length; i++) {
+              let row: any = {};
+              const entry = results.data[i] as Array<any>;
+              for (let j = 0; j < headers.length; j++) {
+                const header = headers[j];
+                const value = entry[j];
+                row = {
+                  ...row,
+                  [typeof header === "number" ? `"${header}"` : header]: value,
+                };
+              }
+              csvJson.push(row);
+            }
+          }
```

Comment:
Let's extract this to a separate file `DatasetParsingService.ts` so that we can have proper unit tests and documentation for it. In the future, someone without context will have a hard time understanding what's happening here. –– Having a separate file with unit tests and comments will be helpful.